src/query/Index.cpp

```diff
@@ -0,0 +1,152 @@
+#include "Index.hpp"
+#include <boost/range/irange.hpp>
+#include "impl/Indexer.hpp"
+#include "logging/LogMacros.hpp"
+#include "utils/Event.hpp"
+
+namespace precice {
+extern bool syncMode;
+namespace query {
+
+precice::logging::Logger Index::_log{"query::Index"};
+
+namespace bg  = boost::geometry;
+namespace bgi = boost::geometry::index;
+
+struct Index::IndexImpl {
+  impl::MeshIndices indices;
+};
+
+Index::Index(const mesh::PtrMesh &mesh)
+    : _mesh(mesh)
+{
+  _pimpl = std::make_unique<IndexImpl>(IndexImpl{});
+}
+
+Index::~Index()
+{
+}
```

Comment:
You can also tell the compiler to provide the default.

```suggestion
Index::~Index() = default;
```

Reply:
Unfortunately, I cant, since I have an incomplete type for pimpl, the compiler complains.